### PR TITLE
Add Twingo III and Zoe 2 to the known vehicle endpoint lists

### PR DESCRIPTION
**Pull request by tmenguy**
Adding two cars to the endpoint list to remove res_state api, and possibly lock status, but need verification from more user.
@epenet  for symmetry I also changed the ChargeStatus so it behave the same way as the PlugStatus, it is all more consistent with the HomeAssistant part too, and to me more robust to Renault changing stuffs....

### CODE REVIEW

File: src/renault_api/kamereon/models.py [hunk 1]
```diff
@@ -457,11 +465,11 @@ def get_charging_status(self) -> Optional[enums.ChargeState]:
...
-        except ValueError as err:
-            # should we return ChargeState.NOT_AVAILABLE?
-            raise exceptions.KamereonException(
-                f"Unable to convert `{self.chargingStatus}` to ChargeState."
-            ) from err
+        except ValueError:
+            _LOGGER.warning(
+                "Unable to convert `%s` to ChargeState.", self.chargingStatus
+            )
+            return None
```
Comment (epenet) [inline src/renault_api/kamereon/models.py:472]: This is unrelated to the new vehicles - so I have moved it to another PR #1570

File: src/renault_api/kamereon/models.py [hunk 2]
```diff
@@ -110,6 +110,14 @@
...
         "soc-levels": None,  # not supported
     },
+    "X102VE": {  # ZOE phase 2
+        "res-state": None,  # not supported
+        # "lock-status": None,  # seems not supported 404, need to verify
+    },
+    "X071VE": {  # TWINGO III
+        "res-state": None,  # not supported
+        # "lock-status": None,  # seems not supported 404, need to verify
+    },
```
Comment (epenet) [inline src/renault_api/kamereon/models.py:120]: Can we keep the model codes alphabetically sorted?
R5E1VE > X071VE > X101VE > X102VE > ...

File: src/renault_api/kamereon/models.py (hunk 1 above)
Comment (tmenguy) [reply src/renault_api/kamereon/models.py]: yes make sense

File: src/renault_api/kamereon/models.py (hunk 2 above)
Comment (tmenguy) [reply src/renault_api/kamereon/models.py:120]: of course

Comment (tmenguy) [reply src/renault_api/kamereon/models.py:120]: but what do we do for teh two lock status it gives a 404 for me on my 2 cars (the Zoe 2 and the twingo III) but is it the same for others?

Comment (epenet) [reply src/renault_api/kamereon/models.py]: I suggest to mark them as "None" and keep the comment

If someone complains they can create a PR to enable

File: src/renault_api/kamereon/models.py [hunk 3]
```diff
@@ -98,6 +98,10 @@
     "R5E1VE": {  # Renault 5 E-TECH
         "charge-schedule": _KCM_ENDPOINTS["charge-schedule"],
     },
+    "X071VE": {  # TWINGO III
+        "res-state": None,  # not supported
+        # "lock-status": None,  # seems not supported 404, need to verify
```
Comment (epenet) [inline src/renault_api/kamereon/models.py:103]:
```suggestion
        "lock-status": None,  # seems not supported (404)
```

File: src/renault_api/kamereon/models.py [hunk 4]
```diff
@@ -110,6 +114,10 @@
         "res-state": None,  # not supported
         "soc-levels": None,  # not supported
     },
+    "X102VE": {  # ZOE phase 2
+        "res-state": None,  # not supported
+        # "lock-status": None,  # seems not supported 404, need to verify
```
Comment (epenet) [inline src/renault_api/kamereon/models.py:119]:
```suggestion
        "lock-status": None,  # seems not supported (404)
```

**Comment by tmenguy** [comment]
ok, have to change the tests too

File: tests/test_renault_vehicle.py [hunk 5]
```diff
@@ -185,7 +185,12 @@ async def test_get_lock_status(
     """Test get_lock_status."""
     fixtures.inject_get_vehicle_details(mocked_responses, "zoe_50.1.json")
     fixtures.inject_get_lock_status(mocked_responses)
-    assert await vehicle.get_lock_status()
+
+    with pytest.raises(
+            EndpointNotAvailableError,
+            match=f"{EndpointNotAvailableError('lock-status','X102VE' )}"
+    ):
+        await vehicle.get_lock_status()
```
[the remark this answers is not in the document]
Comment (epenet) [reply tests/test_renault_vehicle.py]: Or maybe we can add the vehicle fixture from the json sample here:
https://github.com/hacf-fr/renault-api/issues/404#issuecomment-971969619

I'll try to do that now in a preliminary PR

[the remark this answers is not in the document]
Comment (tmenguy) [reply tests/test_renault_vehicle.py]: as you want, was interesting to test the proper exception, but having  a proper 404  fixture for the lock_status is interesting. Here anyway teh goal was also to test that the endpoint in None ...

File: tests/test_renault_vehicle.py [hunk 6]
```diff
@@ -185,7 +185,12 @@ async def test_get_lock_status(
     """Test get_lock_status."""
     fixtures.inject_get_vehicle_details(mocked_responses, "zoe_50.1.json")
```
Comment (epenet) [inline tests/test_renault_vehicle.py:186]:
```suggestion
    fixtures.inject_get_vehicle_details(mocked_responses, "duster.1.json")
```

File: tests/test_renault_vehicle.py (hunk 5 above)
[the remark this answers is not in the document]
Comment (epenet) [reply tests/test_renault_vehicle.py]: I think the best is to adjust the existing test to use "duster.1.json" in this PR

File: tests/test_renault_vehicle.py (hunk 6 above)
Comment (tmenguy) [reply tests/test_renault_vehicle.py:186]: hum as seen above I would probably leave the current test on Zoe to test the fact that th endpoint is not available, and do another one for the duster to properly test the lock status

Comment (tmenguy) [reply tests/test_renault_vehicle.py:195]: added another duster test so we have both

File: tests/test_renault_vehicle.py (hunk 5 above)
[the remark this answers is not in the document]
Comment (tmenguy) [reply tests/test_renault_vehicle.py]: I ended up doing both: doing the duster for the normal case, testing the Zoe to test the missing endpoint exception, tell me if you really don't want to keep this one :)

**Review by epenet — APPROVED** [review]
LGTM 👍

**Comment by tmenguy** [comment]
cool! I let you merge then
